### PR TITLE
IA-3030 handle cryptomining message from SCC

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-3030

Tested in fiab

See example log

```
{"@timestamp":"2021-11-08T14:07:46.894Z","@version":"1","message":"Subscriber Successfully received data: \"{\\n  \\\"notificationConfigName\\\": \\\"organizations/386193000800/notificationConfigs/leonardo-crypto-test\\\",\\n  \\\"finding\\\": {\\n    \\\"name\\\": \\\"organizations/386193000800/sources/6119898994322174789/findings/f3aec71664673d61db90d8efb99e7bac\\\",\\n    \\\"parent\\\": \\\"organizations/386193000800/sources/6119898994322174789\\\",\\n    \\\"resourceName\\\": \\\"//compute.googleapis.com/projects/terra-2d61a51b/zones/us-central1-a/instances/5289438569693667937\\\",\\n    \\\"state\\\": \\\"ACTIVE\\\",\\n    \\\"category\\\": \\\"Execution: Cryptocurrency Mining Combined Detection\\\",\\n    \\\"sourceProperties\\\": {\\n      \\\"threats\\\": [{\\n        \\\"memory_hash_detector\\\": {\\n          \\\"detections\\\": [{\\n            \\\"percent_pages_matched\\\": 0.23588924387646432,\\n            \\\"binary_name\\\": \\\"linux-x86-64_xmrig_6.15.1\\\"\\n          }, {\\n            \\\"binary_name\\\": \\\"linux-x86-64_xmrig_6.15.2\\\",\\n            \\\"percent_pages_matched\\\": 0.7875399361022364\\n          }],\\n          \\\"binary\\\": \\\"XMRig\\\"\\n        }\\n      }, {\\n        \\\"yara_rule_detector\\\": {\\n          \\\"yara_rule_name\\\": \\\"YARA_RULE1\\\"\\n        }\\n      }, {\\n        \\\"yara_rule_detector\\\": {\\n          \\\"yara_rule_name\\\": \\\"YARA_RULE9\\\"\\n        }\\n      }, {\\n        \\\"yara_rule_detector\\\": {\\n          \\\"yara_rule_name\\\": \\\"YARA_RULE10\\\"\\n        }\\n      }, {\\n        \\\"yara_rule_detector\\\": {\\n          \\\"yara_rule_name\\\": \\\"DYNAMIC_YARA_RULE_BFGMINER_2\\\"\\n        }\\n      }]\\n    },\\n    \\\"securityMarks\\\": {\\n      \\\"name\\\": \\\"organizations/386193000800/sources/6119898994322174789/findings/f3aec71664673d61db90d8efb99e7bac/securityMarks\\\"\\n    },\\n    \\\"eventTime\\\": \\\"2021-10-28T17:34:28.678071760Z\\\",\\n    \\\"createTime\\\": \\\"2021-10-28T17:39:31.719Z\\\",\\n    \\\"severity\\\": \\\"HIGH\\\",\\n    \\\"canonicalName\\\": \\\"projects/1089695574439/sources/6119898994322174789/findings/f3aec71664673d61db90d8efb99e7bac\\\",\\n    \\\"findingClass\\\": \\\"THREAT\\\"\\n  },\\n  \\\"resource\\\": {\\n    \\\"name\\\": \\\"//compute.googleapis.com/projects/test-vk-proj/zones/us-central1-a/instances/5289438569693667937\\\",\\n    \\\"project\\\": \\\"//cloudresourcemanager.googleapis.com/projects/1089695574439\\\",\\n    \\\"projectDisplayName\\\": \\\"terra-test-a3f597e0\\\",\\n    \\\"parent\\\": \\\"//cloudresourcemanager.googleapis.com/projects/1089695574439\\\",\\n    \\\"type\\\": \\\"google.compute.Instance\\\",\\n    \\\"folders\\\": [{\\n      \\\"resourceFolder\\\": \\\"//cloudresourcemanager.googleapis.com/folders/710468670182\\\",\\n      \\\"resourceFolderDisplayName\\\": \\\"CommunityWorkbench\\\"\\n    }, {\\n      \\\"resourceFolder\\\": \\\"//cloudresourcemanager.googleapis.com/folders/617814117274\\\",\\n      \\\"resourceFolderDisplayName\\\": \\\"prod\\\"\\n    }],\\n    \\\"displayName\\\": \\\"saturn-de753eea-4721-42df-90ac-8e5734cd8fd9\\\"\\n  }\\n}\"\nattributes {\n  key: \"googclient_deliveryattempt\"\n  value: \"1\"\n}\nmessage_id: \"3274013759751674\"\npublish_time {\n  seconds: 1636380465\n  nanos: 594000000\n}\n.","logger_name":"org.broadinstitute.dsde.workbench.leonardo.http.Boot","thread_name":"io-compute-3","severity":"INFO","level_value":20000,"traceId":"None"}
{"@timestamp":"2021-11-08T14:07:46.975Z","@version":"1","message":"{\"response\":\"None\",\"result\":\"Succeeded\"}","logger_name":"org.broadinstitute.dsde.workbench.leonardo.http.Boot","thread_name":"io-compute-3","severity":"INFO","level_value":20000,"traceId":"None","googleCall":"com.google.cloud.compute.v1.InstancesClient.deleteInstance(terra-test-a3f597e0, us-central1-a, saturn-de753eea-4721-42df-90ac-8e5734cd8fd9)","duration":"59","result":"Succeeded"}
```
---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/NonLeoMessageSubscriber.scala [hunk 1]
```diff
@@ -123,37 +129,47 @@ class NonLeoMessageSubscriber[F[_]](gkeAlg: GKEAlgebra[F],
...
+    runtimeName: RuntimeName,
+    zone: ZoneName,
+    message: String
+  )(implicit ev: Ask[F, AppContext]): F[Unit] =
+    for {
+      ctx <- ev.ask
+      runtimeInfo <- clusterQuery
+        .getActiveClusterRecordByName(googleProject, runtimeName)
+        .transaction
+      _ <- runtimeInfo.traverse { runtime =>
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/NonLeoMessageSubscriber.scala]: Should we maybe log something or send a metric if we can't find the runtime in the Leo DB? Just thinking the SCC alerts may potentially pick up non-Leo VMs.

**Review by rtitle — APPROVED** [review]
Looks good. I see we already have metrics per message type, would be good to look at `CryptoMining` and `CryptoMiningScc` counts when this is in prod.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/NonLeoMessageSubscriber.scala (hunk 1 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/NonLeoMessageSubscriber.scala]: oh, that's a good point...I'll add a log message for now..think we can extract metrics from log message as well

**Comment by Qi77Qi** [comment]
rkernalspec all failed due to the runtime didnt start in time...I've seen that in develop branch as well...seems sth is going on there...but the projects have been deleted, so I can't really see more than what's in leo log

jenkins retest

**Pull request merged**